DRAFT.md

```diff
@@ -160,6 +158,12 @@ logical state nor to the logical state of other object.
 is default constructible. Moving a value type onto the free store should not add
 or remove the ability to be default constructed.
 
+Note that, due to the requirement to support incomplete `T` types, the `indirect<T>`
```

Comment:
Did you explore defining the default-constructor using something like:
```
template<typename U = T>
indirect() requires std::constructible_from<U>;
```

this would allow defering the computation of whether the T is default-constructible until it’s actually used/referenced.

e.g. https://godbolt.org/z/q6dEn3Wr9

You get into trouble if anyone actually asks that question while T is incomplete.
i.e. asks `std::constructible_from<indirect<T>>` while T is incomplete (it will report false, even if T later turns out to have a default-constructor once defined).
If you uncomment the static-assert in the godbolt example above then it makes the program ill-formed - not because the static_assert fails, but because it forces the compiler to compute the fact that the default constructor is not present, then later when you define the T and try to use indirect<T> default constructor, there is none defined.

Attempts to odr-use the default constructor while the T is incomplete will always be a compile-error, however.

Reply:
@jbcoe just checking - did you see this comment (you merged a few minutes after I wrote it)
do you have any thoughts on this alternative approach?

Reply:
I added https://github.com/jbcoe/value_types/issues/199